.expeditor/scripts/generate_bootstrap_bundle.sh

```diff
@@ -0,0 +1,140 @@
+# of the archive. This manifest file will also be uploaded into
+# S3. Additionally, the file `LATEST` in the S3 bucket will always be
+# a copy of the most recent manifest. This can be used as a "pointer"
+# to find the latest tar artifact.
+#
+# This generates a tar file (not tar.gz; further compression doesn't
+# remove much, given that everything inside is already compressed
+# anyway) that has the following internal structure:
+#
+# |-- ARCHIVE_ROOT
+# |   |-- artifacts
+# |   |   `-- all the hart files
+# |   |-- bin
+# |   |   `-- hab
+# |   `-- keys
+# |       `-- all the origin keys
+#
+
+set -euo pipefail
+
+log() {
+  >&2 echo "BOOTSTRAP LOG: $1"
+}
+
+# This bit of magic strips off the Habitat header (first 6 lines) from
+# the compressed tar file that is a core/hab .hart, and extracts the
+# contents of the `bin` directory only, into the ${dir}
+# directory.
+#
+# Note that `dir` should exist before calling this function.
+extract_hab_binaries_from_hart() {
+    local hart="${1}"
+    local dir="${2}"
+
+    tail --lines=+6 "${hart}" | \
+        tar --extract \
+            --directory="${dir}" \
+            --xz \
+            --strip-components=7 \
+            --wildcards "hab/pkgs/core/hab/*/*/bin/"
+}
+
+# Helper function for running s3 cp with appropriate settings.
+s3_cp() {
+    local src="${1}"
+    local dest="${2}"
+    aws --profile habitat \
+        s3 cp "${src}" "${dest}" \
+        --acl public-read
+}
+
+# This is where we ultimately put all the things in S3.
+readonly s3_bucket="habitat-builder-bootstrap"
+
+# This will be form the name of the archive and manifest files, as
+# well as the directory into which the hart files are downloaded.
+readonly this_bootstrap_bundle="hab_builder_bootstrap_$(date +%Y%m%d%H%M%S)"
+
+########################################################################
+
+sandbox_dir="${this_bootstrap_bundle}"
+log "Downloading packages into ${sandbox_dir}"
+mkdir "${sandbox_dir}"
+
+hab pkg download \
+    --download-directory="${sandbox_dir}" \
+    --file=".expeditor/builder_seed.txt" \
+    --target=x86_64-linux
+
+########################################################################
+
+# Since the whole point of this archive is to have something
+# self-contained from which to bootstrap an entirely new Builder
+# environment, we'll also need access to a `hab` binary.
+#
+# Conveniently, we have just downloaded a hart file for that. To make
+# things easy on us, we can extract the stand-alone `hab` binary from
+# that hart file (this assumes Linux packages, naturally) and store it
+# at `bin/hab` in the tar archive.
+
+# The regex is to add the most general placeholders for "version" and
+# "release" in the hart name, while also avoiding similarly-named
+# packages like
+# `core-hab-launcher-${VERSION}-${RELEASE}-x86_64-linux.hart`, etc.
+hab_hart=$(find "${sandbox_dir}"/artifacts -type f -regex '.*/core-hab-[^-]*-[^-]*-x86_64-linux.hart')
```

Comment:
Does this run after builder-worker is built and promoted, and if not, will this always return a single hart that is the latest? 

What I'm thinking here is that if builder-worker hasn't been built and promoted before this runs, we could end up with two versions of `core/hab` in the artifacts path.  `habitat/builder-worker` deps on `core/hab-studio` which deps on `core/hab-backline` which deps on `core/hab-plan-build` which deps on `core/hab`.

Reply:
Ugh, that is a very good catch :man_facepalming: 

Once we get the builder-worker incorporated into the pipeline (https://github.com/habitat-sh/builder/issues/1217), it won't be an issue. In the meantime, we can probably just grab the Habitat `manifest.json` file to get a concrete `hab` version. Alternatively, we could sort the results of this `find` and grab the "biggest" one.

I'll play around and see what looks best.

Reply:
Addressed in https://github.com/habitat-sh/builder/pull/1245/commits/b55f30367d193e3ba1e6c2ef4e17b5c9145b4303